src/utils/metrics/tags.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{config::Config, utils::releases};
+use itertools::Itertools;
+use regex::Regex;
+use std::collections::HashMap;
+
+pub(super) struct NormalizedTags {
```

Comment:
I think I would just call this struct something like `MetricsTags`, then have a function defined on the struct which normalizes all the tags